packages/react/src/components/ActionList/ActionListContext.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, useContext, useMemo } from 'react';
+
+export type onActionEvent =
+  | React.MouseEvent<HTMLLIElement | HTMLAnchorElement>
+  | React.KeyboardEvent<HTMLLIElement | HTMLAnchorElement>;
+
+export type onActionCallbackFunction = (
+  key: string,
+  event: onActionEvent
+) => void;
+
+export type ActionListSelectionType = 'single' | 'multiple';
+
+type ActionListContext = {
+  role: 'list' | 'menu' | 'listbox' | null;
+  selectionType: ActionListSelectionType | null;
+  onAction?: onActionCallbackFunction;
+};
+
+type ActionListProviderProps = {
+  children: React.ReactNode;
+} & ActionListContext;
+
+/* istanbul ignore next */
+const ActionListContext = createContext<ActionListContext>({
+  role: null,
+  selectionType: null,
+  onAction: () => null
+});
+
+function ActionListProvider({
+  role,
+  selectionType,
+  onAction,
+  children
+}: ActionListProviderProps): React.JSX.Element {
+  const { Provider } = ActionListContext as React.Context<ActionListContext>;
```

Comment:
Does this need to be cast, the type is defined above `createContext<ActionListContext>`

Reply:
I don't remember why this was done, but looks like this is safe to remove.